### ISSUE TITLE
mAP is Zero always in train.py

### ISSUE THREAD

**Issue by wefantasy** [ultralytics/yolov5#7617]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello,
I'm trying to train yolov5 using coco128 dataset.
I have already setup the dataset folder manually with mini_coco128.yaml.

When I train the yolov5 model (yolov5s, yolov5m, yolov5n...) I have mAP and labels equal to zero during training(and the training process is extremely fast):
```
python train.py  --epochs 110 --batch-size 8 --workers 0  --weights D:\CommonWorkspace\yolov5\models\yolov5m.pt ^
    --cfg D:\ProgramDownloads\coco128\coco128\yolov5m.yaml ^
    --data D:\ProgramDownloads\coco128\coco128\mini_coco128.yaml ^
    --project D:\Program downloads\coco128\coco128\runs\train
```
![(WKNT6QIVWEG_5NH59A6_XS](https://user-images.githubusercontent.com/32964778/165678222-e66c7b73-ec57-4112-b954-e5d2dcc4389f.png)

Some invalid attempts:
- change cuda version: cuda11.6, cuda11.5, cuda10.2
- change pytorch version: 1.11.0, 1.9.0, 1.5.0
- change dataset: coco128, celeba

Relevant issues don't solve my problems: [#5862](https://github.com/ultralytics/yolov5/issues/5862), [#5902](https://github.com/ultralytics/yolov5/issues/5862), [#1642](https://github.com/ultralytics/yolov5/issues/1642)

Someone has an idea why? 

### Additional

- YOLO: YOLOV5=v6.1 torch=1.11.0+cu113 torchaudio=0.11.0+cu113 torchvision=0.12.0+cu113
- OS: Windows 10 Nvidia GeForce 1660 super
- CUDA: 11.6
- Python: 3.9.0
- Dataset：[coco128](https://www.kaggle.com/datasets/ultralytics/coco128)

**Comment by glenn-jocher** [comment]
@wefantasy it appears you may have environment problems. Please ensure you meet all dependency requirements if you are attempting to run YOLOv5 locally.  If in doubt, create a new virtual [Python 3.9](https://www.python.org/) environment, clone the latest repo ([code changes daily](https://github.com/ultralytics/yolov5/commits/master)), and `pip install` [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) again from scratch.

💡 ProTip! Try one of our verified environments below if you are having trouble with your local environment.


## Requirements

[**Python>=3.7.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.7**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

[Models](https://github.com/ultralytics/yolov5/tree/master/models) and [datasets](https://github.com/ultralytics/yolov5/tree/master/data) download automatically from the latest YOLOv5 [release](https://github.com/ultralytics/yolov5/releases) when first requested.

## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab and Kaggle** notebooks with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>


## Status

![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), validation ([val.py](https://github.com/ultralytics/yolov5/blob/master/val.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/export.py)) on MacOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by wefantasy** [comment]
@glenn-jocher Thank you for your reply, I will try these methods in the future

**Comment by Ashlesha1995** [comment]
Use latest version of repo and while training custom data, follow this link 
https://docs.ultralytics.com/yolov5/tutorials/train_custom_data

**Comment by diangu001** [comment]
hello, boys
in val.py
from
class_map = coco80_to_coco91_class() if is_coco else list(range(1000))
to
class_map = coco80_to_coco91_class() # if is_coco else list(range(1000))

**Comment by wefantasy** [comment]
> hello, boys in val.py from class_map = coco80_to_coco91_class() if is_coco else list(range(1000)) to class_map = coco80_to_coco91_class() # if is_coco else list(range(1000))

![train log](https://user-images.githubusercontent.com/32964778/166853063-63d0a41e-71d3-4e3c-9428-9624ed119c5e.png)
Thank you for your reply. It doesn't seem to solve my problem. I think it's probably the software version, because the MAP is not 0 under certain specific versions, but there will be other problems. What is puzzling is that other pytorch programs can run successfully in this environment.

**Comment by glenn-jocher** [comment]
@wefantasy Thank you for your feedback. It sounds like you continue to face challenges despite the modification. It's likely that the issue is related to the software environment. Have you tried using a different software environment with specific versions of PyTorch and other dependencies for YOLOv5? Often, specific combinations of software versions can affect the success of training. If you encounter any other issues, please don't hesitate to ask.
